lib/rester/client.rb

```diff
@@ -54,15 +55,22 @@ def _path_with_version(path)
     end
 
     def _process_response(path, status, body)
-      if status.between?(200, 299)
-        _parse_json(body)
+      response = Response.new(status)
+
+      if response.successful?
+        response.merge!(_parse_json(body))
       elsif status == 400
-        raise Errors::RequestError, _parse_json(body)[:message]
+        response[:error] = Errors::RequestError
+        response[:message] = _parse_json(body)[:message]
       elsif status == 404
-        raise Errors::NotFoundError, "#{path}"
+        response[:error] = Errors::NotFoundError
+        response[:message] = "#{path}"
       else
-        raise Errors::ServerError, _parse_json(body)[:message]
+        response[:error] = Errors::ServerError
+        response[:message] = _parse_json(body)[:message]
       end
+
```

Comment:
We should be raising for server errors (50x) but not request errors (40x).

Also, we shouldn't be changing the data that the service responds with.